File: day2/part1.py
# Read file input
with open("input.txt", "r") as f:
	instructions = [(str(i.split(" ")[0]), int(i.split(" ")[1])) for i in f]
	
pos_horiz = 0
pos_vert = 0
for instruction in instructions:
	if instruction[0] == "forward":
		pos_horiz += instruction[1]
	elif instruction[0] == "down":
		pos_vert += instruction[1]
	elif instruction[0] == "up":
		pos_vert -= instruction[1]

pos_product = pos_horiz * pos_vert
print(f"Result: {pos_product}")

File: day2/part1.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
	freopen("input.txt", "r", stdin);
	string instruction_name;
	int instruction_value;

	int pos_horiz = 0;
	int pos_vert = 0;
	while (cin >> instruction_name >> instruction_value) {
		if (instruction_name == "forward") {
			pos_horiz += instruction_value;
		}
		else if (instruction_name == "down") {
			pos_vert += instruction_value;
		}
		else if (instruction_name == "up") {
			pos_vert -= instruction_value;
		}
	}
	
	int pos_product = pos_horiz * pos_vert;
	cout << "Result: " << pos_product << '\n';
}
